src/plugins/intel_npu/src/plugin/npuw/util.cpp

```diff
@@ -111,6 +111,47 @@ void unpack_nf4f16(const ov::SoPtr<ov::ITensor>& from,
     }
 }
 
+void unpack_f8f16(const ov::SoPtr<ov::ITensor>& from,
+                  const ov::SoPtr<ov::ITensor>& scale,
+                  const ov::SoPtr<ov::ITensor>& to,
+                  const ov::npuw::util::UnpackOptions& unpack_options) {
+    auto from_shape = from->get_shape();
+    auto scale_shape = scale->get_shape();
+
+    NPUW_ASSERT(from->is_continuous());
+    NPUW_ASSERT(to->is_continuous());
+    NPUW_ASSERT(scale->is_continuous());
+    NPUW_ASSERT(from->get_size() == to->get_size());
+    NPUW_ASSERT(from_shape[0] == scale_shape[0]);
```

Comment:
you should also check `scale_shape[0]` to be `1` to ensure it is a per-channel coefficient?

Reply:
It's [32k x 1]. Works fine and is the same as nf4 unpack

Reply:
ok then you need to theck `scale_shape[1] == 1` as defines your traversal order over the scale tensor.

NF4/INT4 doesn't matter much here, it is the per-channel  unpack that's what important

Reply:
The assert is already there 1 line below